### ISSUE TITLE
Accepted Answer issues

### ISSUE THREAD

**Issue by BWPanda** [backdrop-ops/forum.backdropcms.org#81]
I don't like the accepted answer as it currently stands:

- It doesn't list any author, date, etc. info
- It's duplicated down with the rest of the other answers

I propose the following:

- Add the author, date, etc. to the accepted answer (make it exactly the same as other comments, apart from the different BG colour)
- Remove the accepted answer from the main comment list to avoid duplication
- Add an option, that users can change, to display comments by:
    - Accepted/helpful (I think there's another issue that mentions this too...): comments would be listed as the accepted answer first, then the rest in order of their 'helpful' rating, ignoring date altogether
    - Date: comments would be listed in date order only (accepted and helpful ones can still be marked as such, but that wouldn't affect their ordering)

**Comment by laryn** [comment]
In the date-sorted view, there could still be a notification at the top that says "this topic has an accepted answer" with an anchor link down to it.

**Comment by BWPanda** [comment]
I no longer think my initial ideas are any good.

I now believe we need to make a decision about how comments are ordered on the page:
1. In date order
2. In 'helpful' order

Option 1 is the default in Backdrop for comments, and it allows you to read from top to bottom in the order comments were posted. It also means people can reply to other comments and reading through the list makes sense (and they're displayed in a threaded list).

Option 2 is more like what StackExchange does - users can vote for 'helpful' answers, or up/down vote any answer, and then all comments/answers are ordered by most helpful/up-voted, to least. This, however, means that you can't reply to other comments, as reading through the list of comments would then be out-of-order and not make sense. But it also means the most helpful comments are at the top.

The current state of the forum is a combination of the two, and I don't believe it works. The 'accepted' answer is displayed first, but with no indication of who posted it, or when. Then helpful answers are displayed next, but alos with no indication of the author or date. Finally you have the date-ordered list of all comments, but with no indication of which of those are the accepted or helpful answers.

I don't think I'm the only one who, when reading the 'accepted' answer for a topic, wants to read it in context. Often they reference other comments, so it's hard to read it on its own.

So I think we should decide whether we want a discussion forum (option 1) where we have date-ordered, threaded comments, or a question/answer-type forum where each answer is its own thing and they're ordered by popularity.

**Comment by klonos** [comment]
My 2c:

- Order by date (current default): we could have it so that the following happen:
  - The post title is updated, to include a `[RESOLVED]` suffix
  - The comment that was chosen as "accepted answer" by the post author gets a special style, to make it stand out when scrolling through the rest of the comments.
  - Instead of the duplicated comment, we have a "jump to accepted answer" link that automatically scrolls people to the comment.

- Order by comment number of "upvotes": I like this option, but it should still mimic SE, where each comment still has threaded replies that "follow" it.

**Comment by BWPanda** [comment]
@klonos I agree re. the order by date suggestions. If we decide to stick with a discussion forum rather than a simple question/answer system, then I prefer those suggestions over the current implementation.

**Comment by laryn** [comment]
I'd like to retain the ability to mark an answer as accepted (and provide a quick way to get to it) and perhaps theme the "accepted" and "helpful" answers slightly differently to stand out. I like @klonos suggestions for "Order by date" and while I like SE's model, we'd need more major changes to mimic it more closely in order to be effective (counting/displaying upvotes, improving the UI for this type of system, etc.)

So I'd be fine with either but since "Order by date" seems much more achievable I probably lean that way.
